pkg/controller/statusmanager/status_manager.go

```diff
@@ -37,6 +36,9 @@ type StatusManager struct {
 	name    string
 	version string
 
+	degradedMutex sync.Mutex
```

Comment:
As a general rule, using multiple mutexes increases the chance of a deadlock. Unless you really need that fine-grained access control, I'd recommend just using a single one.

Reply:
The reason I did that was because `setDegraded` is called from "all over the place" (`controller.go`, `operconfig_controller.go`, `clusterconfig_controller.go`), which - with only one mutex - risks generating a high thread contention. `setDegraded` never invokes a lock on `resourceMutex`, so a deadlock should thus be avoided.

Reply:
> risks generating a high thread contention

It's called from lots of places, but it's not called especially *frequently*. I don't think there will be a problem with only one mutex

Reply:
Fair enough, I will update my PR